src/index.js

```diff
@@ -54,11 +54,20 @@ module.exports = class BlobbyClient extends EventEmitter {
   }
 
   getFile(storage, fileKey, opts) {
+    const $client = this;
     return new Promise((resolve, reject) => {
       storage.fetch(fileKey, opts, (err, headers, data) => {
-        if (err) return void reject(err);
-
-        resolve([headers, data]);
+        if (!err) return void resolve([headers, data]);
+        const { statusCode } = err;
+        const encodedFileKey = fileKey.split('/').map(encodeURIComponent).join('/');
+        if (encodedFileKey === fileKey || ![403, 404].includes(statusCode)) {
+          return void reject(err);
+        }
+        storage.fetch(encodedFileKey, opts, (err1, headers1, data1) => {
+          if (err1) return void reject(err1);
+          resolve([headers1, data1]);
+          putFile($client, storage, fileKey, { headers: headers1, buffer: data1 }, { contentType: headers1.ContentType } );
```

Comment:
please add `.catch(() => null)` since we don't want promise rejections to cause issue.